### PR TITLE
Docca reference

### PR DESCRIPTION
**Pull request by alandefreitas**
Preview:

Main: https://alandefreitas.github.io/boost-utility-docs/

Reference: https://alandefreitas.github.io/boost-utility-docs/utility/quickref.html

### CODE REVIEW

**Comment by alandefreitas** [comment]
@vinniefalco I sent you a link to the branch but forgot to tag you here.

Almost all issues unrelated to docca have already been fixed. We're working on those docca issues now.

File: throw_exception.html [hunk 1]
```diff
@@ -7,7 +7,7 @@
 <body>
 Automatic redirection failed, please go to
 <a href="../throw_exception/doc/html/throw_exception.html#using_boost_throw_exception">../throw_exception/doc/html/throw_exception.html#using_boost_throw_exception</a>.&nbsp;<hr>
-<p>� Copyright Beman Dawes, 2001</p>
+<p>Copyright © Beman Dawes, 2001</p>
```
Comment (vinniefalco) [inline throw_exception.html:10]: does the copyright symbol need to be escaped?

**Comment by vinniefalco** [comment]
Where's the doc preview?

**Comment by alandefreitas** [comment]
> Where's the doc preview?

https://alandefreitas.github.io/boost-utility-docs/

http://github.com/alandefreitas/boost-utility-docs

**Comment by vinniefalco** [comment]
Two of the same name:
image

**Comment by vinniefalco** [comment]
This thing
image

File: throw_exception.html (hunk 1 above)
Comment (alandefreitas) [reply throw_exception.html:10]: I have seen `©`, `(c)`, and `&copy;` in the original files.

That's what this one looked like:

https://github.com/boostorg/utility/blob/05e0d1688dcfcd3fdd65bdb6884b7edd1ad2c93c/throw_exception.html#L10

I've been mostly trying to leave them alone or use the closest template I can find. Of course, I do make mistakes sometimes.

**Comment by vinniefalco** [comment]
I don't think we want `B` to appear in the template parameter list, since it names a detail type. I could be wrong though, this needs investigation:
image

**Comment by vinniefalco** [comment]
This "This Page Intentionally Left Blank" thing has to be fixed, not sure how. Maybe playing with the auto-generated table of contents. Or writing our own table of contents.

**Comment by vinniefalco** [comment]
The layout of the reference looks very nice!

**Comment by alandefreitas** [comment]
> Two of the same name:

That must have happened when I was packing the reference. I'll push that commit in a minute. 😮

> This thing

There's an [issue](https://github.com/boostorg/docca/issues/106) for this one already. We're working on all the docca issues in parallel to this. That's the other [issue](https://github.com/boostorg/docca/issues/105). They might be even related.

> I don't think we want B to appear in the template parameter list, since it names a detail type. I could be wrong though, this needs investigation:

Yes. I have avoided changing any header files. I think it'll be easier to get this specific PR accepted if we leave the Javadoc to another PR, once there's some consensus on the reference itself. Including the Javadoc stuff in this PR would involve changing almost all files in Boost.Utility and the only consensus I noticed so far is to not mess with what's working 😆 . 

> This "This Page Intentionally Left Blank" thing has to be fixed, not sure how. Maybe playing with the auto-generated table of contents. Or writing our own table of contents.

I haven't opened an issue with that yet but we have been already discussing and thinking of alternatives. We haven't really found a solution in that other discussion (in the PR). But I can see some alternatives, such as one section per component or hiding some sections as headings.

It seems like the biggest related problem right now is the "up arrow". We need to be able to customize where the "up arrow" links to so that any of the solutions above work properly. 

We've been discussing that too. I'll open the issue as soon as I understand more precisely what and where the issue is.

**Comment by alandefreitas** [comment]
> There's an issue for this one already

By the way, that's the other [issue](https://github.com/boostorg/docca/issues/105). They might be even related.

**Comment by alandefreitas** [comment]
Since Doxygen picks up the first declaration it finds rather than the definition, shouldn't [things](https://alandefreitas.github.io/boost-utility-docs/utility/ref/boost__string_view.html) like 

image

really say "**Declared** in header <boost/utility/string_view_fwd.hpp>"?

**Comment by vinniefalco** [comment]
Yes you have to use `#ifdef BOOST_UTILITY_DOCS` and `#ifndef BOOST_UTILITY_DOCS` to control what doxygen "sees".

**Comment by alandefreitas** [comment]
> Yes you have to use #ifdef BOOST_UTILITY_DOCS and #ifndef BOOST_UTILITY_DOCS to control what doxygen "sees".

Wow... I thought that didn't apply to forward declarations. That's worse than I thought then. 😂

**Comment by alandefreitas** [comment]
> This "This Page Intentionally Left Blank" thing has to be fixed, not sure how. Maybe playing with the auto-generated table of contents. Or writing our own table of contents.

I've been experimenting with that. If we remove the blank page, we have this:

image

which is unacceptable. Besides, the up arrow still leads to the second blank "reference" page:

image

So there are a few imperfect solutions I've been thinking of. I managed to fix the issue with two reference sections and the up arrow with:

```
[section:ref Reference]
[xinclude quickref.xml]
[include reference.qbk]
[endsect]
```

image

So the only problem left is hiding the subsections from the table of contents. 

The solutions proposed so far are:

1. Docca would create a small number of parent sections of some name at a higher level. Something like [this](https://www.boost.org/doc/libs/1_77_0/libs/log/doc/html/index.html) or [this](https://www.boost.org/doc/libs/1_77_0/doc/html/intrusive.html).
2. Docca would be able to create headings instead of sections. Everything would go to a single HTML file.
3. Move the reference one level deeper and limit the depth to 2 levels for the whole TOC. Something like [this](https://www.boost.org/doc/libs/1_77_0/doc/html/interprocess.html). 
4. Somehow create one docca reference per component. This would probably lead to other problems that would require some new features in docca so the new links would work.
5. Docca would generate docbook directly in order to control chunk.section.depth
6. Abandon docca

Any solution that is very specific is probably going to cause trouble as they ask us to change more specific details. Some other solutions would require us to include many new features in docca.

So I've been leaning towards moving the reference to a deeper level as it's the least intruvise option. I just couldn't find the command to set the maximum TOC depth level in quickbook yet.

image

On the other hand, the up arrow is now working, which is nice.

**Comment by vinniefalco** [comment]
>  I just couldn't find the command to set the maximum TOC depth level in quickbook yet.

You set that in the Jamfile:
https://github.com/boostorg/json/blob/54397e66056ab8148ea8906809ee0f1707fc9f37/doc/Jamfile#L103

**Comment by alandefreitas** [comment]
Oh! It's an option and not a command then. 

The TOC looks good [now](https://alandefreitas.github.io/boost-utility-docs/index.html).

image

Unfortunately, it doesn't only affect the TOC. It also stops generating links to the proper sections at level 3. The [reference links](https://alandefreitas.github.io/boost-utility-docs/utility/index_and_ref/ref.html) go to the home page now. 

I've noticed the documentation of most boost libraries I've seen put the complete reference on a single page. I'm starting to think this is not by design.

I'll keep trying.

**Comment by vinniefalco** [comment]
You can disable the ToC completely and write it manually, like asio does. Then you can avoid the "this page intentionally left blank"

**Pull request merged**